File: app/src/main/java/ohi/andre/consolelauncher/managers/music/Song.java
package ohi.andre.consolelauncher.managers.music;

import java.io.File;

import it.andreuzzi.comparestring2.StringableObject;

/**
 * Created by francescoandreuzzi on 17/08/2017.
 */

public class Song implements StringableObject  {

    private long id;
    private String title, path, lowercaseTitle, singer;

    public Song(long songID, String songTitle, String singer) {
        id = songID;
        title = songTitle;
        this.singer = singer;
        this.lowercaseTitle = title.toLowerCase();
    }

    public Song(File file) {
        String name = file.getName();
        int dot = name.lastIndexOf(".");
        if (dot != -1) {
            name = name.substring(0, dot);
        }

        this.title = name;
        this.path = file.getAbsolutePath();
        this.id = -1;
        this.singer = "Unknown";
        this.lowercaseTitle = title.toLowerCase();
    }

    public long getID() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getSinger() {
        return singer;
    }

    public String getPath() {
        return path;
    }

    @Override
    public String getLowercaseString() {
        return lowercaseTitle;
    }

    @Override
    public String getString() {
        return title;
    }
}

File: app/src/main/java/ohi/andre/consolelauncher/managers/music/Song.kt
package ohi.andre.consolelauncher.managers.music

import it.andreuzzi.comparestring2.StringableObject
import java.io.File

class Song : StringableObject {
    private var id: Long
    private var title: String
    private var path: String? = null
    private var lowercaseTitle: String
    private var singer: String
    private var source: String = MusicService.SOURCE_INTERNAL
    private var podcastShowId: String? = null
    private var podcastEpisodeKey: String? = null
    private var startPositionMs: Int = 0

    constructor(songID: Long, songTitle: String, singer: String) {
        id = songID
        title = songTitle
        this.singer = singer
        lowercaseTitle = title.lowercase()
    }

    constructor(file: File) {
        var name = file.name
        val dot = name.lastIndexOf(".")
        if (dot != -1) {
            name = name.substring(0, dot)
        }

        title = name
        path = file.absolutePath
        id = -1
        singer = "Unknown"
        lowercaseTitle = title.lowercase()
    }

    constructor(
        title: String,
        singer: String,
        path: String,
        source: String,
        podcastShowId: String?,
        podcastEpisodeKey: String?,
        startPositionMs: Int
    ) {
        id = -1
        this.title = title
        this.singer = singer
        this.path = path
        this.source = source
        this.podcastShowId = podcastShowId
        this.podcastEpisodeKey = podcastEpisodeKey
        this.startPositionMs = startPositionMs
        lowercaseTitle = title.lowercase()
    }

    fun getID(): Long = id

    fun getTitle(): String = title

    fun getSinger(): String = singer

    fun getPath(): String? = path

    fun getSource(): String = source

    fun getPodcastShowId(): String? = podcastShowId

    fun getPodcastEpisodeKey(): String? = podcastEpisodeKey

    fun getStartPositionMs(): Int = startPositionMs

    override fun getLowercaseString(): String = lowercaseTitle

    override fun getString(): String = title
}
